Set titleLabel's backgroundColor on tintColorDidChange

natanrolnik opened the pull request:
When initializing a `TORoundedButton` in the following way:

```swift
let button = RoundedButton(text: "I agree")
button.tintColor = UIColor.appPurple
button.textColor = .white
button.backgroundColor = .white
button.cornerRadius = 4
```

This is the result:

![RoundedButton](https://user-images.githubusercontent.com/1164565/57652648-b3b6d400-75d8-11e9-8bb2-c18246e18fe3.png)

Tapping on the button makes this glitch go away, because `self.titleLabel.backgroundColor = self.isTapped ? [UIColor clearColor] : self.tintColor;` is called.

This PR sets the `titleLabel`'s `backgroundColor` on `layoutSubviews` to avoid this bug.

TimOliver commented:
Thanks for the PR @natanrolnik!

Hmm, I think I actually had that sort of code set previously, however, it was causing issues with the interactive animation.

Can you try tapping down on the button, sliding your finger off, and then quickly back on again, and see if it flickers?

It definitely looks like I forgot to update the `UILabel` background color at some point. It's usually best to try and avoid setting in `layoutSubviews` as much as possible since it can get called multiple times inside an animation block.

natanrolnik commented:
Makes sense. I'll think about it - maybe it should be set whenever `tintColorDidChange` is called instead? That should solve this issue as well.

natanrolnik commented:
Ok @TimOliver - I've moved the line to `tintColorDidChange` which seems to be a better fit, in this case, considering the animations you mentioned.

I've tested and no flickering happens.

I've updated the PR.

natanrolnik commented:
I made a small mistake related to indenting and empty lines - this is why you see force-pushes. It's good to merge now.

TimOliver commented:
Awesome! Looks perfect to me! Thank you so much for doing that! :D

natanrolnik commented:
Great! 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>
